driver-core/src/test/unit/com/mongodb/MongoCredentialSpecification.groovy

```diff
@@ -192,6 +198,7 @@ class MongoCredentialSpecification extends Specification {
         null == credential.getPassword()
         mechanism == credential.getAuthenticationMechanism()
         MongoCredential.MONGODB_X509_MECHANISM == credential.getMechanism()
+        credential == MongoCredential.createMongoX509Credential()
```

Comment:
Given the names of these test methods (... should populate the correct fields), consider creating a separate test method for equals/hashCode ('equivalent credentials should be equal and have same hashCode').